packages/datetime/src/timePicker.tsx

```diff
@@ -252,10 +274,30 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         this.updateTime(loopTime(newTime, unit), unit);
     }
 
+    private isTimeInRange(date: Date) {
+        const format = "H:mm:ss:SSS";
```

Comment:
why does format matter here? this is just for stringifying but we should be able to compare the actual data itself, not a string representation thereof.